rskj-core/src/test/java/co/rsk/peg/federation/FederationSupportImplTest.java

```diff
@@ -19,164 +19,371 @@
+                .withActivations(activations)
+                .build();
+
+            Optional<Script> activeFederationRedeemScript = federationSupport.getActiveFederationRedeemScript();
+            assertTrue(activeFederationRedeemScript.isPresent());
+            assertThat(activeFederationRedeemScript.get(), is(genesisFederation.getRedeemScript()));
+        }
 
-        assertThat(activeFederation, is(genesisFederation));
+        @Test
+        @Tag("getActiveFederationAddress")
+        void getActiveFederationAddress_returnsGenesisFederationAddress() {
+            Address activeFederationAddress = federationSupport.getActiveFederationAddress();
+            assertThat(activeFederationAddress, is(genesisFederation.getAddress()));
+        }
     }
 
-    @Test
-    void getActiveFederation_withNewFederationAndNullOldFederation_returnsNewFederation() {
-        Federation newFederation = FederationTestUtils.getErpFederation(networkParameters);
-        storageProvider.setNewFederation(newFederation);
+    @Nested
+    @Tag("one null federation")
+    @TestInstance(TestInstance.Lifecycle.PER_CLASS)
+    class ActiveFederationTestsWithOneNullFederation {
+        @BeforeAll
+        void setUp() {
+            // create old and new federations
+            P2shErpFederationBuilder p2shErpFederationBuilder = new P2shErpFederationBuilder();
+            oldFederation = p2shErpFederationBuilder
+                .build();
+            List<BtcECKey> newFederationKeys = BitcoinTestUtils.getBtcEcKeysFromSeeds(
+                new String[]{"fa01", "fa02", "fa03", "fa04", "fa05", "fa06", "fa07", "fa08", "fa09"}, true
+            );
+            newFederation = p2shErpFederationBuilder
+                .withMembersBtcPublicKeys(newFederationKeys)
+                .build();
+
+            StorageAccessor storageAccessor = new InMemoryStorage();
+            storageProvider = new FederationStorageProviderImpl(storageAccessor);
+
+            federationSupport = federationSupportBuilder
+                .withFederationConstants(federationMainnetConstants)
+                .withFederationStorageProvider(storageProvider)
+                .build();
+        }
+
+        @ParameterizedTest
+        @Tag("getActiveFederation")
+        @MethodSource("expectedFederationArgs")
+        void getActiveFederation_returnsExpectedFederation(Federation oldFederation, Federation newFederation, Federation expectedFederation) {
+            storageProvider.setOldFederation(oldFederation);
+            storageProvider.setNewFederation(newFederation);
 
-        Federation activeFederation = federationSupport.getActiveFederation();
+            Federation activeFederation = federationSupport.getActiveFederation();
+            assertThat(activeFederation, is(expectedFederation));
+        }
+
+        private Stream<Arguments> expectedFederationArgs() {
+            return Stream.of(
+                Arguments.of(oldFederation, null, genesisFederation),
```

Comment:
Is this a real scenario? Is it possible to have an old federation and an active federation to be the genesis? That's not realistic. Once you have an old federation, one federation must be active, different than the genesis, because for an old federation exists, the genesis should be changed.